### PR TITLE
 The file change and removed unwanted files

### PR DESCRIPTION
**Pull request by pkshah1006**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-ssm-patchbaseline/.gitignore [hunk 1]
```diff
@@ -12,9 +12,12 @@ out.java
 out/
 .settings
 .project
-
 # auto-generated files
 target/
 
 # our logs
 rpdk.log
```
Comment (dandaji) [inline aws-ssm-patchbaseline/.gitignore:19]: remove those whitespaces, 
if the file doesn't change at all, please remove those unnecessary changes.

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java [hunk 2]
```diff
@@ -122,6 +123,10 @@ private DescribePatchBaselinesResponse describePatchBaselinesResponse(final Stri
                 .stream().map(baseline -> baseline.baselineId()).collect(Collectors.toList());
 
         for (String baselineId : baselineIdList) {
+            if(baselineId.contains(File.separator)){
```
Comment (dandaji) [inline aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:126]: i still don't understand this. 
My argument is that:

baselines returned as an arn  arn:aws:ssm:us-east-2:733109147000:patchbaseline/pb-0e392de35e7c563b7 instead of only baseline id pb-0e392de35e7c563b7, means this baseline is owned by Patch Manager. 

and later it's passed to readHandler to read the content of the baseline, for Patch owned baseline, you need to input the whole arn arn:aws:ssm:us-east-2:733109147000:patchbaseline/pb-0e392de35e7c563b7 , instead of pb-0e392de35e7c563b7 to succeed.

Comment (dandaji) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:126]: So with your current code, the baseline arn is truncated to baseline Id, when it's passed to readHandler, it would definitely fail, because the baselineId doesn't exist.

Comment (dandaji) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:126]: so this means it would fail, i don't understand how did it succeed

File: aws-ssm-patchbaseline/template.yml [hunk 3]
```diff
@@ -19,4 +19,4 @@ Resources:
     Properties:
       Handler: software.amazon.ssm.patchbaseline.HandlerWrapper::testEntrypoint
       Runtime: java8
-      CodeUri: ./target/aws-ssm-patchbaseline-handler-1.0-SNAPSHOT.jar
+      CodeUri: ./target/aws-ssm-patchbaseline-handler-1.0-SNAPSHOT.jar
```
Comment (dandaji) [inline aws-ssm-patchbaseline/template.yml:22]: nothing new changed to this file as well, please undo the change. so it won't show up as new file

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java (hunk 2 above)
Comment (dandaji) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:126]: for example, the following is a patch owned baseline in us-east--1, 
 arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c 

**aws ssm get-patch-baseline --baseline-id pb-03e3f588eec25344c** would run into the following exception: 

An error occurred (DoesNotExistException) when calling the GetPatchBaseline operation: Patch Baseline to be retrieved does not exist.

**aws ssm get-patch-baseline --baseline-id arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c** is the request input

{
    "BaselineId": "arn:aws:ssm:us-east-1:075727635805:patchbaseline/pb-03e3f588eec25344c",
    "Name": "AWS-CentOSDefaultPatchBaseline",
    "OperatingSystem": "CENTOS",
    "GlobalFilters": {
        "PatchFilters": []
    },
    "ApprovalRules": {
        "PatchRules": [
            {
                "PatchFilterGroup": {
                    "PatchFilters": [
                        {
                            "Key": "PRODUCT",
                            "Values": [
                                "*"
                            ]
                        }
                    ]
                },
                "ComplianceLevel": "UNSPECIFIED",
                "ApproveAfterDays": 7,
                "EnableNonSecurity": true
            }
        ]
    },
    "ApprovedPatches": [],
    "ApprovedPatchesComplianceLevel": "UNSPECIFIED",
    "ApprovedPatchesEnableNonSecurity": false,
    "RejectedPatches": [],
    "RejectedPatchesAction": "ALLOW_AS_DEPENDENCY",
    "PatchGroups": [],
    "CreatedDate": 1522345092.515,
    "ModifiedDate": 1522345092.515,
    "Description": "Default Patch Baseline for CentOS Provided by AWS.",
    "Sources": []
}

File: aws-ssm-patchbaseline/template.yml (hunk 3 above)
Comment (dandaji) [reply aws-ssm-patchbaseline/template.yml:22]: readHandler would definitely fail if you stipe down the return baseline Id

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java (hunk 2 above)
Comment (dandaji) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:126]: now i think about it, adding a readHandler inside ListHandler may not be as appropriate, we can return the value of BaselineIdentities returned from describe-patch-baselines directly. 
Can you try this out?

Comment (pkshah1006) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java]: yes we have tired the same.
But if we tried to return "describe-patch-baselines " it doesnot contains the pactchbaseline full details i.e: what you have mentioned above and It will contain the data as below.

DescribePatchBaselinesResponse(BaselineIdentities=[
  PatchBaselineIdentity(BaselineId=arn: aws: ssm: us-east-1: 075727635805: patchbaseline/pb-03e3f588eec25344c,
  BaselineName=AWS-CentOSDefaultPatchBaseline,
  OperatingSystem=CENTOS,
  BaselineDescription=DefaultPatchBaselineforCentOSProvidedbyAWS.,
  DefaultBaseline=true)
])

Since, we are taking the baseline id from describe patch baseline response and passing the baseline id to  getpatchbasline functionality to get the baseline details as you mentioned.
Now the Listhandler process is working as you mentioned.

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java [hunk 4]
```diff
@@ -117,23 +135,27 @@ private DescribePatchBaselinesResponse describePatchBaselinesResponse(final Stri
...
+            GetPatchBaselineRequest getPatchBaselineRequest = GetPatchBaselineRequest.builder()
+                    .baselineId(baselineId)
                     .build();
-            ProgressEvent<ResourceModel, CallbackContext> response
-                    = readHandler.handleRequest(proxy, requestFromId, null, logger);
 
-            ResourceModel resourceModel = response.getResourceModel();
-            models.add(resourceModel);
+            GetPatchBaselineResponse getPatchBaselineResponse =
+                    proxy.injectCredentialsAndInvokeV2(getPatchBaselineRequest, ssmClient::getPatchBaseline);
```
Comment (dandaji) [inline aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:153]: My comment is not to use getPatchBaseline here directly,
My intention is to remove readHandler(getPatchBaseline call) completely from ListHandler, as customer are not expecting to get the full details of each baseline from ListHandler, but to get describe-patch-baselines return value, i.e. BaselineIdentities returned from describe-patch-baselines

Comment (dandaji) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:153]: we will not return the full details.

Comment (pkshah1006) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java]: As you mentioned i have made the changes in Listhandler operation.
Now the ListHandler operation will return  the describePatchBaseline details of BaselineEntities and the response should be like as below

DescribePatchBaselinesResponse(BaselineIdentities=[
  PatchBaselineIdentity(BaselineId=arn: aws: ssm: us-east-1: 075727635805: patchbaseline/pb-03e3f588eec25344c,
  BaselineName=AWS-CentOSDefaultPatchBaseline,
  OperatingSystem=CENTOS,
  BaselineDescription=DefaultPatchBaselineforCentOSProvidedbyAWS.,
  DefaultBaseline=true)
]
]

File: aws-ssm-patchbaseline/.gitignore (hunk 1 above)
Comment (dandaji) [reply aws-ssm-patchbaseline/.gitignore:19]: no changes here, this file should not be in Files Changed.

File: aws-ssm-patchbaseline/aws-ssm-patchbaseline.json [hunk 5]
```diff
@@ -300,13 +300,6 @@
                 "ssm:RegisterDefaultPatchBaseline"
             ]
         },
-        "delete": {
```
Comment (dandaji) [inline aws-ssm-patchbaseline/aws-ssm-patchbaseline.json:303]: no changes here, so this file shouldn't be in Files Changed

File: aws-ssm-patchbaseline/inputs/inputs_1_read.json [hunk 6]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (dandaji) [inline aws-ssm-patchbaseline/inputs/inputs_1_read.json:1]: are inputs_1_read, inputs_1_list, inputs_1_delete json files needed?

File: aws-ssm-patchbaseline/inputs/inputs_1_invalid.json [hunk 7]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (dandaji) [inline aws-ssm-patchbaseline/inputs/inputs_1_invalid.json:1]: please put a correct format of json file as inputs_1_invalid.json content, meaning: the format is still valid, but the input values are not valid.

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java [hunk 8]
```diff
@@ -12,21 +12,37 @@
 import software.amazon.awssdk.services.ssm.SsmClient;
 import software.amazon.ssm.patchbaseline.utils.SsmClientBuilder;
 import software.amazon.awssdk.services.ssm.model.SsmException;
+import software.amazon.awssdk.services.ssm.model.GetPatchBaselineRequest;
```
Comment (dandaji) [inline aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/ListHandler.java:15]: please remove unused imports

File: aws-ssm-patchbaseline/template.yml (hunk 3 above)
Comment (dandaji) [reply aws-ssm-patchbaseline/template.yml:22]: this files shouldn't be in Files Changed, as nothing changed

File: aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/translator/resourcemodel/ReadResourceModelTranslator.java [hunk 9]
```diff
@@ -59,4 +63,30 @@ public static ResourceModel translateToResourceModel(final GetPatchBaselineRespo
         return model;
     }
 
+    /**
+     * Translate Read Response to Resource Model
+     */
+    public static ResourceModel translateToResourceModel(final DescribePatchBaselinesResponse describePatchBaselinesResponse,
```
Comment (dandaji) [inline aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/translator/resourcemodel/ReadResourceModelTranslator.java:69]: what is tags for?

Comment (dandaji) [inline aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/translator/resourcemodel/ReadResourceModelTranslator.java:69]: this is for List, i think you can create a separate file for it, or you change the name of this file

Comment (pkshah1006) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/translator/resourcemodel/ReadResourceModelTranslator.java:69]: Now we have reverted the changes in ReadResourceModelTranslator and created the new file for listhandler resouce model operations

Comment (pkshah1006) [reply aws-ssm-patchbaseline/src/main/java/software/amazon/ssm/patchbaseline/translator/resourcemodel/ReadResourceModelTranslator.java:69]: Removed the tags parameter since we are not using Tags details in describe patch baseline

File: aws-ssm-patchbaseline/template.yml (hunk 3 above)
Comment (pkshah1006) [reply aws-ssm-patchbaseline/template.yml:22]: yes we dont have any changes and we have reverted to previous version

File: aws-ssm-patchbaseline/.gitignore (hunk 1 above)
Comment (pkshah1006) [reply aws-ssm-patchbaseline/.gitignore:19]: We dont have any changes and now we have reverted to the previous stable changes and removed white spaces in the file

File: aws-ssm-patchbaseline/template.yml (hunk 3 above)
Comment (pkshah1006) [reply aws-ssm-patchbaseline/template.yml:22]: The file has reverted to previous stable changes.
